tests/release/release.go

```diff
@@ -0,0 +1,133 @@
+	var managedNamespace string
+
+	var _ = Describe("test-release-service-happy-path", func() {
+		BeforeAll(func() {
+			// Recreate random namespaces names per each test because if using same namespace names, the next test will not be able to create the namespaces as they are terminating
+			devNamespace = "user-" + uuid.New().String()
+			managedNamespace = "managed-" + uuid.New().String()
+
+			// Create the dev namespace
+			_, err := framework.CommonController.CreateTestNamespace(devNamespace)
+			Expect(err).NotTo(HaveOccurred(), "Error when creating namespace '%s': %v", devNamespace, err)
+
+			// Create the managed namespace
+			_, err = framework.CommonController.CreateTestNamespace(managedNamespace)
+			Expect(err).NotTo(HaveOccurred(), "Error when creating namespace '%s': %v", managedNamespace, err)
+		})
+
+		AfterAll(func() {
+			// Delete the dev and managed namespaces with all the resources created in them
+			Expect(framework.CommonController.DeleteNamespace(devNamespace)).NotTo(HaveOccurred())
+			Expect(framework.CommonController.DeleteNamespace(managedNamespace)).NotTo(HaveOccurred())
+		})
+
+		var _ = Describe("Creation of the 'Happy path' resources", func() {
+			It("Create an ApplicationSnapshot.", func() {
+				_, err := framework.ReleaseController.CreateApplicationSnapshot(snapshotName, devNamespace, applicationName, snapshotComponents)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			It("Create Release Strategy", func() {
+				_, err := framework.ReleaseController.CreateReleaseStrategy(releaseStrategyName, managedNamespace, releasePipelineName, releasePipelineBundle, releaseStrategyPolicy, releaseStrategyServiceAccount)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			It("Create ReleasePlan in dev namespace", func() {
+				AutoReleaseLabel := ""
+				_, err := framework.ReleaseController.CreateReleasePlan(sourceReleasePlanName, devNamespace, applicationName, managedNamespace, AutoReleaseLabel)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			It("Create ReleasePlanAdmission in managed namespace", func() {
+				AutoReleaseLabel := ""
+				_, err := framework.ReleaseController.CreateReleasePlanAdmission(targetReleasePlanAdmissionName, devNamespace, applicationName, managedNamespace, AutoReleaseLabel, releaseStrategyName)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			It("Create a Release", func() {
+				_, err := framework.ReleaseController.CreateRelease(releaseName, devNamespace, snapshotName, sourceReleasePlanName)
+				Expect(err).NotTo(HaveOccurred())
+			})
+		})
+
+		var _ = Describe("Post-release verification", func() {
+			It("A PipelineRun should have been created in the managed namespace", func() {
+				Eventually(func() error {
+					_, err := framework.ReleaseController.GetPipelineRunInNamespace(managedNamespace, releaseName, devNamespace)
+
+					return err
+				}, 1*time.Minute, defaultInterval).Should(BeNil())
+			})
+
+			It("The PipelineRun should exist and succeed", func() {
```

Comment:
Doesn't the check before this already check if it exists? The name of the check is to make sure it was created

Reply:
I updated the test name to be "The PipelineRun has started, was done, and succeeded" which is more accurate